lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -823,6 +829,44 @@ def upload_reads(self, ctx, params):
         self.log('Starting upload reads, parsing args')
         o, wsid, name, objid, kbtype, single_end, fwdid, revid, shock = (
             self._proc_upload_reads_params(params))
+
+        needs_to_be_interleaved = 0
+        if not single_end and revid is not None:
+            needs_to_be_interleaved = 1
+
+        if revid and shock:
+            #Download files so we can interleave them later.
+            fileinput = [{'shock_id': fwdid,
+                          'file_path': self.scratch + '/fwd/',
+                          'unpack': 'uncompress'}]
+            fileinput.append({'shock_id': revid,
+                              'file_path': self.scratch + '/rev/',
+                              'unpack': 'uncompress'})
+            dfu = DataFileUtil(self.callback_url)
```

Comment:
move the dfu declaration outside the if block and remove the dfu declaration on line 903, just declare it once